tests/cli/test_schema_generation.py

```diff
@@ -118,7 +120,6 @@ def test_gen_pipeline_schema_no_components(self):
                 ],
                 catch_exceptions=False,
             )
-            assert result.exit_code == 1
```

Comment:
why did you remove the assertion? probably it should just be moved outside the context manager, no?

Reply:
:thinking: true, I don't remember why I removed, probably didn't intend to.

Will bring it back.

Reply:
I read a bit and I think it could be that we don't need that assertion. We expect an exception, so either we get it and the assertion isn't reached or we don't get the exception and the test fails anyway.

Reply:
[See here](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.raises):

> When using pytest.raises as a context manager, it’s worthwhile to note that normal context manager rules apply and that the exception raised must be the final line in the scope of the context manager. Lines of code after that, within the scope of the context manager will not be executed. For example:

```python
>>> value = 15
>>> with pytest.raises(ValueError) as exc_info:
...     if value > 10:
...         raise ValueError("value must be <= 10")
...     assert exc_info.type is ValueError  # this will not execute
Instead, the following approach must be taken (note the difference in scope):
```
```python
>>> with pytest.raises(ValueError) as exc_info:
...     if value > 10:
...         raise ValueError("value must be <= 10")
...
>>> assert exc_info.type is ValueError
```

Reply:
makes sense